core/src/main/java/org/mapfish/print/wrapper/PJoinedArray.java

```diff
@@ -53,32 +53,50 @@ public PArray getArray(final int i) {
 
   @Override
   public int getInt(final int i) {
-    return (Integer) get(i);
-  }
+    Object o = getObjectAt(i);
+    return (Integer) o;
 
+  }
   @Override
   public long getLong(final int i) {
-    return (Long) get(i);
+    Object o = getObjectAt(i);
+    return (Long) o;
   }
 
   @Override
   public float getFloat(final int i) {
-    return (Float) get(i);
+    Object o = getObjectAt(i);
+    return (Float) o;
   }
 
   @Override
   public double getDouble(final int i) {
-    return (Double) get(i);
+    Object o = getObjectAt(i);
+    return (Double) o;
   }
 
   @Override
   public String getString(final int i) {
-    return get(i).toString();
+    return getObjectAt(i).toString();
   }
 
   @Override
   public boolean getBool(final int i) {
-    return (Boolean) get(i);
+    Object o = getObjectAt(i);
+    return (Boolean) o;
+  }
+
+  /**
+   * Created to keep creating a controlled {@link NullPointerException}.
+   * @param index in the array
+   * @return the object or an NPE if index is out of bounds.
+   */
+  private Object getObjectAt(final int index) {
+    Object o = get(index);
+    if (o == null) {
+      throw new NullPointerException("No value found at index:" + index);
```

Comment:
I didn't like to raise a NullPointerException...

Reply:
New RuntimeException as discussed